python/lsst/meas/algorithms/reinterpolate_pixels.py

```diff
@@ -0,0 +1,140 @@
+
+class ReinterpolatePixelsConfig(pexConfig.Config):
+    """Config for ReinterpolatePixelsTask"""
+
+    kernelFwhm = pexConfig.Field[float](
+        doc="FWHM of double Gaussian smoothing kernel.",
+        default=1.0,
+    )
+
+    growFootprints = pexConfig.Field[int](
+        doc="Number of pixels to grow footprints for pixels.",
+        default=1,
+    )
+
+    maskNameList = pexConfig.ListField[str](
+        doc="Mask plane name.",
+        default=["SAT"],
+    )
+
+    fallbackValue = pexConfig.Field[float](
+        doc="Value of last resort for interpolation.",
+        default=None,
+        optional=True,
+    )
+
+
+class ReinterpolatePixelsTask(pipeBase.Task):
+    """Task to reinterpolate pixels"""
+
+    ConfigClass = ReinterpolatePixelsConfig
+    _DefaultName = "reinterpolatePixels"
+
+    def run(self, exposure):
+        """Reinterpolate pixels over ``exposure``.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure to interpolate over.
+
+        Notes
+        -----
+        ``exposure`` is modified in place and will become the reinterpolated exposure.
+        """
+        mask = exposure.mask
+
+        if self.config.growFootprints > 0 and "SAT" in self.config.maskNameList:
+            # If we are interpolating over an area larger than the original
+            # masked region, we need to expand the original mask bit to the
+            # full area to explain why we interpolated there.
+            from lsst.ip.isr import growMasks
+
+            growMasks(mask, radius=self.config.growFootprints, maskNameList=["SAT"], maskValue="SAT")
+
+        from lsst.ip.isr import Defects
+
+        thresh = afwDetection.Threshold(
+            mask.getPlaneBitMask(self.config.maskNameList), afwDetection.Threshold.BITMASK
+        )
+        fpSet = afwDetection.FootprintSet(mask, thresh)
+        defectList = Defects.fromFootprintList(fpSet.getFootprints())
+        self._interpolateDefectList(exposure, defectList)
+
+    def _interpolateDefectList(self, exposure, defectList):
+        """Interpolate over defects specified in a defect list.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure to process.
+        defectList : `lsst.meas.algorithms.Defects`
+            List of defects to interpolate over.
+
+        Notes
+        -----
+        ``exposure`` is modified in place and will become the reinterpolated exposure.
+        """
+
+        psf = self._createPsf(self.config.kernelFwhm)
+        if self.config.fallbackValue is None:
+            fallbackValue = afwMath.makeStatistics(exposure.image, afwMath.MEANCLIP).getValue()
+        else:
+            fallbackValue = self.config.fallbackValue
+        if "INTRP" not in exposure.mask.getMaskPlaneDict():
+            exposure.mask.addMaskPlane("INTRP")
+        maskedImage = MaskedImageF(image=exposure.image, mask=exposure.mask, variance=exposure.variance)
```

Comment:
Instead of making a new `MaskedImage` here, I think you can just pass `exposure.maskedImage` to `interpolateOverDefects`, and then you won't have to assign to `exposure.image` etc. afterwards.

Reply:
Done!